w3af/plugins/crawl/find_backdoors.py

```diff
@@ -118,27 +118,30 @@ def _check_if_exists(self, web_shell_url):
         except BaseFrameworkException:
             om.out.debug('Failed to GET webshell:' + web_shell_url)
         else:
-            signature = self._match_signature(response)
-            if signature is None:
-                return
+            if response.get_code() == 200:
+                signature = self._match_signature(response)
+                if signature is None:
+                    return
```

Comment:
Ok, now I understand the root cause! Thanks for explaining it to me.

I just created this issue: https://github.com/andresriancho/w3af/issues/15103 to track and fix the problem.

This PR will be closed without merging since, even though it does fix the issue, also potentially reduces the situations in which a backdoor is found.